src/components/dialog/dialog.js

```diff
@@ -1,107 +1,293 @@
+ * parameters passed in.
+ * @param {object} settings
  */
-
-import {keyCodes, defaultClassNames} from '../../constants';
-
-
 const UIDialog = ({
         dialog = '.js-dialog',
         openBtn = '.js-dialog-btn',
         closeBtn = '.js-dialog-close-btn',
         isModal = false,
+        isAlert = false,
+        readyClass = defaultClassNames.IS_READY,
+        activeClass = defaultClassNames.IS_ACTIVE,
         showBackdrop = true
     } = {}) => {
 
-    let DOM,
-        state = {};
+    // Stores all the dom nodes for the module
+    let DOM = {
+        dialogs: qa(dialog)
+    };
+
+    // Keeps track of current state
+    let state = {
+        currentOpenButton: null,
+        currentDialog: null,
+        focusableElements: null //  elements within modal
+    };
 
-    // Create the backdrop
-    const backdrop = document.createElement('div');
 
-    backdrop.classList.add(defaultClassNames.DIALOG_BACKDROP);
+    /**
+     * @function bindBackdropEvents
+     * @desc Adds event listener for clicking on backdrop
+     */
+    function bindBackdropEvents() {
+        DOM.backdrop.addEventListener('click', handleBackdropClick);
+    }
 
-    init();
 
     /**
-     * @function init
-     * @desc Initialises the dialog
+     * @function unbindBackdropEvents
+     * @desc Removes event listener for clicking on backdrop
      */
-    function init() {
+    function unbindBackdropEvents() {
+        DOM.backdrop.removeEventListener('click', handleBackdropClick);
+    }
 
-        // Save all DOM queries for future use
-        DOM = {
-            'page':     document.querySelectorAll('body')[0],
-            'dialog':   document.querySelectorAll(dialog)[0],
-            'openBtn':  document.querySelectorAll(openBtn)[0],
-            'closeBtn': document.querySelectorAll(closeBtn)[0]
-        };
 
-        // Check if the dialog exists, return if not
-        if (DOM.dialog === undefined) {
-            return false;
+    /**
+     * @function handleBackdropClick
+     * @desc If backdrop has been clicked, dismiss dialog
+     * @param {Event} e
+     */
+    function handleBackdropClick(e) {
+        if (e.target === DOM.backdrop) {
+            hideDialog(state.currentDialog);
         }
+    }
+
+
+    /**
+     * @function createBackdrop
+     * @desc Creates the dialog backdrop
+     */
+    function createBackdrop() {
+        // Create the backdrop
+        DOM.backdrop = createEl({id: 'hawk', className: defaultClassNames.DIALOG_BACKDROP});
+
+        addBackdropA11y(DOM.backdrop);
+    }
 
-        // Remove backdrop if turned off
-        if (!showBackdrop) {
-            DOM.dialog.classList.add(defaultClassNames.NO_BACKDROP);
+
+    /**
+     * @function addBackdropA11y
+     * @desc Applies relevant roles and attributes to the backdrop
+     * @param {node} backdrop
+     */
+    function addBackdropA11y(backdrop) {
+        backdrop.setAttribute('aria-hidden', true);
+    }
```

Comment:
Pretty sure we don't need this at all as the backdrop is just an empty `<div>` which just gets ignored?